salt/react_apps.sls

```diff
@@ -25,39 +29,75 @@ include:
       - pkg: git
       - user: {{ entry.user }}_user_exists
 
-{{ appdir }}:
-  file.symlink:
-    - target: {{ directory }}
-    - makedirs: True
-    - runas: {{ entry.user }}
-    - require:
-      - git: {{ entry.git.url }}
-    # The symlink must be created before Apache starts.
-    - require_in:
-      - service: apache2
-
 {% if 'apache' in entry %}
 {{ apache(entry.git.target, entry.apache, context=context) }}
 {% endif %}
 
-{{ userdir }}-nvm-installer:
-  file.managed:
-    - name: {{ userdir }}/nvm-install.sh
-    - source: https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh
-    # curl -sS https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh | shasum
-    - source_hash: e70e16f272f4ce4cc3e8d98e1e0ead449ab9ae5c
-    - user: {{ entry.user }}
-    - group: {{ entry.user }}
-    - mode: 755
+{{ directory }}/.env:
+  file.copy:
+    - source: {{ directory }}/.env.salt
+
+{{ userdir }}-yarn-install:
+  cmd.run:
+    - name: yarn
+    - runas: {{ entry.user }}
+    - cwd: {{ directory }}
+    - require:
+      - pkg: git
+    - onchanges:
+      - git: {{ entry.git.url }}
 
-{{ userdir }}-nvm-install:
+{{ userdir }}-yarn-build:
   cmd.run:
-    - name: bash {{ userdir }}/nvm-install.sh
+    - name: yarn build
+    - env: {{ entry.env|yaml }}
     - runas: {{ entry.user }}
-    - cwd: {{ userdir }}
+    - cwd: {{ directory }}
     - require:
       - pkg: git
     - onchanges:
-      - file: {{ userdir }}-nvm-installer
+      - git: {{ entry.git.url }}
+
+{% if salt['pillar.get']('ver_txt:enable') %}
+{{ build_asset_dir }}/ver.txt:
+  file.managed:
+    - contents: "branch: {{ entry.git.branch }} || commit_hash: {{ salt['cmd.shell']('cd ' + directory + '&& git rev-parse --verify HEAD') }} || time: {{ timestamp }}"
+    - onchanges:
+      - cmd: {{ userdir }}-yarn-build
+{% endif %}{# ver txt #}
+
+{{ directory }}/.htaccess:
```

Comment:
Wouldn't it be much simpler to manage the .htaccess contents in this repository instead of in covid-19-procurement-explorer-public ? That way, we can also include the prerender token using Jinja templating.

Reply:
i couldn't determine the appropriate location for .htaccess file in deploy repo

Reply:
I put it in covid19.include.jinja